src/services/offer.js

```diff
@@ -239,6 +239,16 @@ class OfferService {
         return offer;
     }
 
+    async getOfferByCompanyID(companyID, user, hasAdminPrivileges) {
+        let offers = await Offer.find({ owner: companyID });
+
+        offers = offers.filter((offer) =>
+            !(offer.isHidden && !(hasAdminPrivileges || offer.owner.toString() === user?.company?._id.toString()))
+        );
+
+        return offers;
+    }
```

Comment:
```suggestion
        return Offer.find({ owner: companyID })
        .filter((offer) =>
            !(offer.isHidden && !(hasAdminPrivileges || offer.owner.toString() === user?.company?._id.toString()))
        );
    }
```

Reply:
I've tried this and it results in an error:
UNEXPECTED ERROR: TypeError: Offer.find(...).filter is not a function

Reply:
Maybe (await Offer.find(...)).filter(...)

Reply:
Yep, `.find` returns a promise that resolves to an array